File: Java/src/StringProblems/P1.java
package StringProblems;

// Program to count words of a string

public class P1 {
    public static int countWords(String str){
        if(str == null || str.trim().isEmpty()){
            return 0;
        }
        String[] words = str.trim().split("\\s+");
        return words.length;
    }
    public static void main(String[] args) {
        String str = "My name is Mishun Nair.";  
        int wordCount = countWords(str);
        System.out.println("The number of words in given string is: " + wordCount);
    }
}

File: TSPractice/StringProblems/P1.ts
// Program to count words of a string

const string1: string = "My name is Midhun Nair and I am an engineer.";
function countWords(str: string): number {
    if(!str.trim()) {
        return 0;
    }
    const words: string[] = str.trim().split(/\s+/);
    return words.length;
}
console.log("Number of words in the given string is " + countWords(string1));